pyocd/target/family/target_lpc5500.py

```diff
@@ -75,6 +83,29 @@ def _modify_ap1(self, seq):
         
         return seq
 
+    def check_locked_state(self, seq):
+        # Attempt to unlock cores if they are locked (flash is empty etc.)
+        self.was_core_locked = False
+        if not DM_AP in self.aps:
+            exists = coresight.ap.AccessPort.probe(self.dp, DM_AP)
+            if not exists:
+                LOG.error("Error: no DMAP exists")
+                return
+            ap_addr = coresight.ap.APv1Address(DM_AP)
+            ap = coresight.ap.AccessPort.create(self.dp, ap_addr)
+            self.unlock()
```

Comment:
Can you pass this `ap` object to `unlock()` so it is used instead of doing low level AP/DP register accesses in `unlock()`?

Reply:
Please see updated commit